tracelib/trace.js

```diff
@@ -41,6 +41,14 @@ function parseInfo(info) {
   return info;
 }
 
+let streamingCallbacks = [];
+function pushEvent(e) {
+  events.push(e);
+  for (let {callback, predicate} of streamingCallbacks) {
+    if (!predicate || predicate(e)) callback(e);
```

Comment:
can these callbacks be async? (so that their costs don't get added to other execution traces)

Reply:
Made them async. Thanks. (They were batched and delivered async later on in the callstack, but I think your point still holds).